platform/darwin/src/MGLShapeSource.mm

```diff
@@ -56,6 +56,21 @@ - (instancetype)initWithIdentifier:(NSString *)identifier shape:(nullable MGLSha
     return self;
 }
 
+- (instancetype)initWithIdentifier:(NSString *)identifier features:(NS_ARRAY_OF(id <MGLFeature>) *)features options:(nullable NS_DICTIONARY_OF(MGLShapeSourceOption, id) *)options {
+    for (id <MGLFeature> feature in features) {
+        if (![feature conformsToProtocol:@protocol(MGLFeature)]) {
+            [NSException raise:NSInvalidArgumentException format:@"The object %@ included in the features argument does not conform to the MGLFeature protocol.", feature];
+        }
+    }
+    MGLShapeCollectionFeature *shapeCollectionFeature = [MGLShapeCollectionFeature shapeCollectionWithShapes:(NSArray *)features];
```

Comment:
Since `MGLShapeCollectionFeature` inherits from `MGLShapeCollection` and `MGLShapeCollection` uses an `NS_ARRAY_OF(MGLShape *)` I don't think I can change `MGLShapeCollectionFeature` since it would require the change to bubble up to `MGLShapeCollection` 😬 

The implementation of `MGLShapeCollectionFeature` does not use lightweight generics (just a plain old NSArray). I made the same change to `MGLShapeSource` in https://github.com/mapbox/mapbox-gl-native/pull/7665/commits/6287c6bc3bbf6c7df7653c84318b01f026e910f5. Are you ok with this solution for this case, for now?